R/defineModel.R

```diff
@@ -1,18 +1,40 @@
-                        constraints=c("cases","none","items"),std.err=c("quick","full","none"), distribution=c("normal","discrete"), method=c("gauss", "quadrature", "montecarlo", "quasiMontecarlo"),
-                        n.iterations=2000,nodes=NULL, p.nodes=2000, f.nodes=2000,converge=0.001,deviancechange=0.0001, equivalence.table=c("wle","mle","NULL"), use.letters=FALSE,
-                        allowAllScoresEverywhere = TRUE, guessMat = NULL, est.slopegroups = NULL, fixSlopeMat = NULL, slopeMatDomainCol=NULL, slopeMatItemCol=NULL, slopeMatValueCol=NULL,
-                        progress = FALSE, Msteps = NULL, increment.factor=1 , fac.oldxsi=0, export = list(logfile = TRUE, systemfile = FALSE, history = TRUE, covariance = TRUE, reg_coefficients = TRUE, designmatrix = FALSE) )   {
+defineModel <- function(dat, items, id, splittedModels = NULL,
+                        irtmodel = c("1PL", "2PL", "PCM", "PCM2", "RSM", "GPCM", "2PL.groups", "GPCM.design", "3PL"),
+                        qMatrix=NULL, DIF.var=NULL, HG.var=NULL, group.var=NULL,
+                        weight.var=NULL, anchor = NULL, domainCol=NULL, itemCol=NULL,
+                        valueCol=NULL,check.for.linking = TRUE, minNperItem = 50,
+                        removeMinNperItem = FALSE, boundary = 6, remove.boundary = FALSE,
+                        remove.no.answers = TRUE, remove.no.answersHG = TRUE,
+                        remove.missing.items = TRUE, remove.constant.items = TRUE,
+                        remove.failures = FALSE, remove.vars.DIF.missing = TRUE,
+                        remove.vars.DIF.constant = TRUE, verbose=TRUE, software = c("conquest","tam"),
+                        dir = NULL, analysis.name, schooltype.var = NULL, model.statement = "item",
+                        compute.fit = TRUE, pvMethod = c("regular", "bayesian"),
+                        fitTamMmlForBayesian = TRUE, n.plausible=5, seed = NULL,
+                        conquest.folder=NULL, constraints=c("cases","none","items"),
+                        std.err=c("quick","full","none"), distribution=c("normal","discrete"),
+                        method=c("gauss", "quadrature", "montecarlo", "quasiMontecarlo"),
+                        n.iterations=2000, nodes=NULL, p.nodes=2000, f.nodes=2000,
+                        converge=0.001, deviancechange=0.0001, equivalence.table=c("wle","mle","NULL"),
+                        use.letters=FALSE, allowAllScoresEverywhere = TRUE, guessMat = NULL,
+                        est.slopegroups = NULL, fixSlopeMat = NULL, slopeMatDomainCol=NULL,
+                        slopeMatItemCol=NULL, slopeMatValueCol=NULL, progress = FALSE,
+                        Msteps = NULL, increment.factor=1 , fac.oldxsi=0,
+                        export = list(logfile = TRUE, systemfile = FALSE, history = TRUE,
+                                      covariance = TRUE, reg_coefficients = TRUE, designmatrix = FALSE) ){
```

Comment:
arguments with checks:

- [ ] dat
- [ ] items
- [ ] id
- [ ] splittedModels
- [ ] irtmodel
- [ ] qMatrix
- [ ] DIF.var
- [ ] HG.var
- [ ] group.var
- [ ] weight.var
- [ ] anchor
- [ ] domainCol
- [ ] itemCol
- [ ] valueCol
- [ ] check.for.linking
- [ ] minNperItem
- [ ] removeMinNperItem
- [ ] boundary
- [ ] remove.boundary
- [ ] remove.no.answers
- [ ] remove.no.answersHG
- [ ] remove.missings.items
- [ ] remove.constant.items
- [ ] remove.failures
- [ ] remove.vars.DIF.missing
- [ ] remove.vars.DIF.constant
- [ ] verbose
- [ ] software
- [ ] dir
- [ ] analysis.name
- [ ] schooltype.var
- [ ] model.statement
- [ ] compute.fit
- [ ] pvMethod
- [ ] fitTamMmlForBayesian
- [ ] n.plausible
- [ ] seed
- [ ] conquest.folder
- [ ] constraints
- [ ] std.err
- [ ] distribution
- [ ] method
- [ ] n.iterations
- [ ] nodes
- [ ] p.nodes
- [ ] f.nodes
- [ ] converge
- [ ] deviancechange
- [ ] equivalence.table
- [ ] use.letters
- [ ] allowAllScoresEverywhere
- [ ] guessMat
- [ ] est.slopegroups
- [ ] fixSlopeMat
- [ ] sloteMatDomainCol
- [ ] slopeMatItemCol
- [ ] slopeMatValuesCol
- [ ] progress
- [ ] Msteps
- [ ] increment.factor
- [ ] fac.oldxsi
- [ ] export